### PR TITLE
fixes false positive for no-unknown animations

### LINKED ISSUE #1771: 'false positive' on no-unknown-animations
The is a bug in the **no-unknown-animations** rule

> What CSS is needed to reproduce this issue?

e.g.

```css
@keyframes fb-FadeIn {
    0% {
        opacity: 0;
    }
    100% {
        opacity: 1;
    }
}

.fb-Test_Item {
    animation: fb-FadeIn .3s linear .7s forwards;
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    
    "no-unknown-animations": true
  }
}
```

> Which version of stylelint are you using?

7.1.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

PostCSS plugin

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

no

> What did you expect to happen?

not to receive a warning

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
test.css
11:26	⚠  Unexpected unknown animation name ".3s" (no-unknown-animations) [stylelint]
11:30	⚠  Unexpected unknown animation name ".7s" (no-unknown-animations) [stylelint]
```

### PR DESCRIPTION
**Pull request by sendilkumarn**
fixes #1771

### CODE REVIEW

File: src/utils/getUnitFromValueNode.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ export default function (node) {
     // ignore hack unit
     .replace("\\0", "")
     .replace("\\9", "")
+    .replace(".", "")
```
Comment (evilebottnawi) [inline src/utils/getUnitFromValueNode.js]: Be good add test to getUnitFromValueNode-test.js (`.1s`).

**Comment by sendilkumarn** [comment]
@evilebottnawi  👍

File: src/utils/getUnitFromValueNode.js (hunk 1 above)
Comment (jeddy3) [inline src/utils/getUnitFromValueNode.js]: Can we get a comment above this line please e.g. `// ignore decimal place`

**Comment by jeddy3** [comment]
LGTM

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 - Added: `--report-needless-disables` and `reportNeedlessDisables` option.
 - Added: `--ignore-disables` and `ignoreDisables` option.
 - Added: `value-list-max-empty-lines` rule.
+- Fixed: `no-unknown-animations` and `unit-blacklist` now handles a number without a leading zero.
```
Comment (davidtheclark) [inline CHANGELOG.md]: Let's say "now handle numbers without leading zeros"

**Comment by davidtheclark** [comment]
One comment about phrasing, otherwise fine by me. Is the source of the problem here that `postcss-value-parser` doesn't understand decimal values without leading zeros? If so, an issue should probably be filed over there, as well.

**Comment by sendilkumarn** [comment]
@davidtheclark I do agree. we have to file an issue there.

**Comment by sendilkumarn** [comment]
@davidtheclark why it is in `needs revision` ?  Did I miss anything ?

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,7 @@
 - Added: `--report-needless-disables` and `reportNeedlessDisables` option.
 - Added: `--ignore-disables` and `ignoreDisables` option.
 - Added: `value-list-max-empty-lines` rule.
+- Fixed: `no-unknown-animations` and `unit-blacklist` now handle numbers without leading zero.
```
Comment (davidtheclark) [inline CHANGELOG.md]: leading zeros (plural)

**Comment by davidtheclark** [comment]
@sendilkumarn Just two outstanding comments to address, I think. Looking good.

File: CHANGELOG.md (hunk 3 above)
Comment (sendilkumarn) [inline CHANGELOG.md]: ideally it is one zero right ?

**Comment by davidtheclark** [comment]
LGTM

**Pull request merged**